compiler/optimizer/src/main/java/org/apache/nemo/compiler/optimizer/NemoOptimizer.java

```diff
@@ -54,11 +58,17 @@
   /**
    * @param dagDirectory to store JSON representation of intermediate DAGs.
    * @param policyName the name of the optimization policy.
+   * @param environmentType the environment type of the workload to optimize the DAG for.
+   * @param clientRPC the RPC channel to communicate with the client.
    */
   @Inject
   private NemoOptimizer(@Parameter(JobConf.DAGDirectory.class) final String dagDirectory,
-                        @Parameter(JobConf.OptimizationPolicy.class) final String policyName) {
+                        @Parameter(JobConf.OptimizationPolicy.class) final String policyName,
+                        @Parameter(JobConf.EnvironmentType.class) final String environmentType,
+                        final ClientRPC clientRPC) {
     this.dagDirectory = dagDirectory;
+    this.environmentType = environmentType;
```

Comment:
`environmentType = EnvironmentType.valueOf(environmentTypeStr)`

Reply:
Do we have to add environmentType? Why do we need this variable and when is it used? Actually, the environment type isn't important because the behavior depends on the optimization policy?

Reply:
It's been refactored to meet with the changes above.